packages/holocron/__tests__/holocronModule.spec.jsx

```diff
@@ -22,20 +22,153 @@ import { fromJS } from 'immutable';
+    });
+    it('should call load if present with props', () => {
+      fakeProps = {
+        load: jest.fn(),
+        myFakeProp: true,
+      };
+      executeLoad(fakeProps);
+      expect(fakeProps.load.mock.calls).toMatchSnapshot();
+    });
+  });
+
+  describe('executeLoadModuleData', () => {
+    it('should call loadModuleData if dispatch and loadModuleData are present', async () => {
+      expect.assertions(2);
+      await executeLoadModuleData(
+        fakeLoadModuleData,
+        FakeComponent,
+        { dispatch: fakeDispatch }
+      );
+      expect(fakeLoadModuleData.mock.calls).toMatchSnapshot();
+      expect(fakeProps.dispatch).toHaveBeenCalled();
+    });
+  });
+
+  describe('executeLoadingFunctions', () => {
+    it('should have setState called with loaded when successful', async () => {
+      expect.assertions(1);
+      await executeLoadingFunctions({
+        loadModuleData: fakeLoadModuleData,
+        WrappedComponent: FakeComponent,
+        frozenProps: fakeProps,
+        loadCount: 0,
+        componentName: 'FakeComponent',
+        hocInstance: fakeInstance,
+      });
+      expect(fakeSetState).toHaveBeenCalledWith({ status: 'loaded' });
+    });
+    it('should not have setState called if successful but over loadCount', async () => {
+      expect.assertions(1);
+      await executeLoadingFunctions({
+        loadModuleData: fakeLoadModuleData,
+        WrappedComponent: FakeComponent,
+        frozenProps: fakeProps,
+        loadCount: 0,
+        componentName: 'FakeComponent',
+        hocInstance: {
+          ...fakeInstance,
+          mounted: false,
+          state: {
+            loadCount: 9999,
+          },
+        },
+      });
+      expect(fakeSetState).not.toHaveBeenCalled();
+    });
+    it('should have setState called with error when failure occurs', async () => {
+      expect.assertions(1);
+      fakeLoadModuleData = () => {
+        throw new Error('Failed');
+      };
+      await executeLoadingFunctions({
+        loadModuleData: fakeLoadModuleData,
+        WrappedComponent: FakeComponent,
+        frozenProps: fakeProps,
+        loadCount: 0,
+        componentName: 'FakeComponent',
+        hocInstance: fakeInstance,
+      });
+      expect(fakeSetState).toHaveBeenCalledWith({ status: 'error' });
+    });
+    it('should not have setState called with error if mounted is false', async () => {
+      expect.assertions(1);
+      fakeLoadModuleData = () => {
+        throw new Error('Failed');
+      };
+      await executeLoadingFunctions({
+        loadModuleData: fakeLoadModuleData,
+        WrappedComponent: FakeComponent,
+        frozenProps: fakeProps,
+        loadCount: 0,
+        componentName: 'FakeComponent',
+        hocInstance: {
+          ...fakeInstance,
+          mounted: false,
+        },
+      });
+      expect(fakeSetState).not.toHaveBeenCalled();
+    });
+  });
+
+  it('should wrap module if no arguments are provided', () => {
```

Comment:
We allow the case of `MyModule.holocron = undefined` as in the past we didn't have to include `holocronModule`. We no longer throw if `name` isn't provided however, if a reducer is supplied, we must supply a `name` and the `reducer`.